reporter/src/main/kotlin/model/EvaluatedModelMapper.kt

```diff
@@ -549,13 +541,7 @@ internal class EvaluatedModelMapper(private val input: ReporterInput) {
         val result = mutableListOf<T>()
 
         values.forEach { value ->
-            val existingValue = find { it == value }
-            if (existingValue != null) {
-                result += existingValue
-            } else {
-                add(value)
-                result += value
-            }
+            result += find { it == value } ?: value.also { add(it) }
```

Comment:
You could go one step further and simplify the whole function to (and maybe also make it an expression function):

```kotlin
return values.map { value -> find { it == value } ?: value.also { add(it) } }.distinct()
```

Reply:
Thanks for the idea, I was able to simplify it even further by calling `addIfRequired(value: T)`.